Userland/Utilities/markdown-check.cpp

```diff
@@ -233,7 +236,13 @@ int main(int argc, char** argv)
     for (auto const& file_item : files) {
         auto file_lexical_path = LexicalPath(file_item.key);
         auto file_dir = file_lexical_path.dirname();
-        for (auto const& file_link : file_item.value.file_links()) {
+        for (auto const& maybe_file_link : file_item.value.file_links()) {
+            if (!maybe_file_link.has_value()) {
+                // This issue was already reported by the link visitor, nothing to do here.
+                any_problems = true;
+                continue;
+            }
+            auto const& file_link = maybe_file_link.value();
```

Comment:
What's the point of this? file_links is never randomly accessed, always iterated, so instead of storing extra data for each entry, as well as adding extra logic for each entry, let's just fail as soon as encounter one.

Reply:
The point was to disturb the current architecture as little as possible, but I guess this is my free ticket to completely rework it :^). However, the "continuing to check other links" is valuable as it can tell you multiple errors at once. Just like sometimes compilers continuing after an error is valuable.

Reply:
Sure, but there are nicer ways to do it then adding random holes into a vector :^)